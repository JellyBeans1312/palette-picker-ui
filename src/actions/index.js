export const addProject = id => ({
  type: 'ADD_PROJECT',
  id
});

export const addAllProjects = allProjects => ({
  type: 'ADD_ALL_PROJECTS',
  allProjects
});

export const saveColors = colors => ({
  type: 'SAVE_COLORS',
  colors
});