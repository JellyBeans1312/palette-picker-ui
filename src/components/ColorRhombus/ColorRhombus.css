.color-rhombus {
  height: 80%;
  width: 16rem;
  transform: skew(-20deg);
  border: 1px solid black;
}