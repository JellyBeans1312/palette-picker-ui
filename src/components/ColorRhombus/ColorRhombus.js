import React from 'react';
import { lockColor } from '../../actions';
import { connect } from 'react-redux';
import './ColorRhombus.css';
import PropTypes from 'prop-types';

export const ColorRhombus = (props) => {
  const color = props.color
  return (
    <div className='color-rhombus' style={{ backgroundColor: props.color }}>
      <button onClick={() => props.lockColor(color)}>Lock</button>
    </div>
  )
}

export const mapDispatchToProps = dispatch => ({
  lockColor: color => dispatch(lockColor(color))
});

export default connect(null, mapDispatchToProps)(ColorRhombus);

ColorRhombus.propTypes = {
  color: PropTypes.string,
  lockColor: PropTypes.func
}