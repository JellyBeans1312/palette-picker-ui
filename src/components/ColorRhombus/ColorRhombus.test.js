import React from 'react';
import { shallow } from 'enzyme';
import { ColorRhombus } from './ColorRhombus';
import { mapDispatchToProps } from './ColorRhombus';
import { lockColor } from '../../actions'

describe('Color Rhombus', () => {
  it('should match snapshot', () => {
    const wrapper = shallow (<ColorRhombus 
      props={{
        color: '#fff',
        lockColor: jest.fn()
      }}
    />)

    expect(wrapper).toMatchSnapshot()
  });
  describe('mapDispatchToProps', () => {
    it('should call action lock color', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = lockColor('#fff');
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.lockColor('#fff');
  
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
});