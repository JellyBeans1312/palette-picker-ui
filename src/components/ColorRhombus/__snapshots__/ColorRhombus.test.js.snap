// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Rhombus should match snapshot 1`] = `
<div
  className="color-rhombus"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <button
    onClick={[Function]}
  >
    Lock
  </button>
</div>
`;
